server/src/test/java/org/spine3/server/SubscriptionServiceShould.java

```diff
@@ -0,0 +1,291 @@
+        subscriptionObserver.checkFields();
+
+        // Activate subscription
+        final MemoizeStreamObserver<SubscriptionUpdate> activationObserver = new MemoizeStreamObserver<>();
+        subscriptionService.activate(subscriptionObserver.streamFlowValue, activationObserver);
+
+        // Post update to Stand directly
+        final ProjectId projectId = ProjectId.newBuilder().setId("some-id").build();
+        final Message projectState = Project.newBuilder().setId(projectId).build();
+        boundedContext.getStand().update(projectId, AnyPacker.pack(projectState));
+
+        // isCompleted set to false since we don't expect activationObserver::onCompleted to be called.
+        activationObserver.checkFields(false);
+    }
+
+    @Test
+    public void cancel_subscription_on_topic() {
+        final BoundedContext boundedContext = setupBoundedContextForAggregateRepo();
+
+        final SubscriptionService subscriptionService = SubscriptionService.newBuilder()
+                                                                           .addBoundedContext(boundedContext)
+                                                                           .build();
+
+        final String type = boundedContext.getStand()
+                                          .getAvailableTypes()
+                                          .iterator()
+                                          .next()
+                                          .getTypeName();
+
+        final Target target = Target.newBuilder()
+                                    .setType(type)
+                                    .build();
+
+        final Topic topic = Topic.newBuilder()
+                                 .setTarget(target)
+                                 .build();
+
+        // Subscribe
+        final MemoizeStreamObserver<Subscription> subscribeObserver = new MemoizeStreamObserver<>();
+        subscriptionService.subscribe(topic, subscribeObserver);
+
+        // Activate subscription
+        final MemoizeStreamObserver<SubscriptionUpdate> activateSubscription = spy(new MemoizeStreamObserver<SubscriptionUpdate>());
+        subscriptionService.activate(subscribeObserver.streamFlowValue, activateSubscription);
+
+        // Cancel subscription
+        subscriptionService.cancel(subscribeObserver.streamFlowValue, new MemoizeStreamObserver<Response>());
+
+        // Post update to Stand
+        final ProjectId projectId = ProjectId.newBuilder().setId("some-other-id").build();
+        final Message projectState = Project.newBuilder().setId(projectId).build();
+        boundedContext.getStand().update(projectId, AnyPacker.pack(projectState));
+
+        // The update must not be handled by the observer
+        verify(activateSubscription, never()).onNext(any(SubscriptionUpdate.class));
+        verify(activateSubscription, never()).onCompleted();
+    }
+
+    private static BoundedContext newBoundedContext(String name) {
```

Comment:
Please see `TestBoundedContextFactory`. I think this should go there (maybe, refactored a bit).